packages/main/src/MenuItem.ts

```diff
@@ -427,8 +539,15 @@ class MenuItem extends ListItem implements IMenuItem {
 		this.fireDecoratorEvent("close");
 	}
 
-	get isMenuItem(): boolean {
-		return true;
+	_updateCheckedState() {
+		if (this._itemCheckMode === ItemCheckMode.None) {
+			return;
+		}
+
+		const newState = !this.checked;
+
+		this.fireDecoratorEvent("item-check");
```

Comment:
The event should be fired after the state has been updated. When the event is triggered, its target must reflect the component’s current state (i.e., checked).

Note: If this event becomes cancelable in the future, a revert mechanism will need to be implemented.

Reply:
This is already obsolete as the event is made private now.